### ISSUE TITLE
App crashes when attached ListView is scrolled on Android 2.3

### ISSUE THREAD

**Issue by reknih** [jamesmontemagno/FloatingActionButton-for-Xamarin.Android#1]
The app will halt twice if in debug mode, the Stack Trace is:

```
[MonoDroid] Java.Lang.NoSuchMethodError: Exception of type 'Java.Lang.NoSuchMethodError' was thrown.
[MonoDroid] at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <IL 0x00011, 0x00047>
[MonoDroid] at Android.Runtime.JNIEnv.GetMethodID (intptr,string,string) [0x00066] in /Users/builder/data/lanes/monodroid-mlion-monodroid-4.21-series/9e05e39f/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:162
[MonoDroid] at Android.Views.View.Animate () [0x00014] in /Users/builder/data/lanes/monodroid-mlion-monodroid-4.21-series/9e05e39f/source/monodroid/src/Mono.Android/platforms/android-21/src/generated/Android.Views.View.cs:10297
[MonoDroid] at com.refractored.fab.FloatingActionButton.Toggle (bool,bool,bool) [0x00053] in c:\GitHub\FloatingActionButton-for-Xamarin.Android\library\FloatingActionButton.cs:210
[MonoDroid] at com.refractored.fab.FloatingActionButton.Hide (bool) [0x00000] in c:\GitHub\FloatingActionButton-for-Xamarin.Android\library\FloatingActionButton.cs:170
[MonoDroid] at com.refractored.fab.FloatingActionButton/AbsListViewScrollDetectorImpl.OnScrollUp () [0x00000] in c:\GitHub\FloatingActionButton-for-Xamarin.Android\library\FloatingActionButton.cs:464
[MonoDroid] at com.refractored.fab.AbsListViewScrollDetector.OnScroll (Android.Widget.AbsListView,int,int,int) [0x0006e] in c:\GitHub\FloatingActionButton-for-Xamarin.Android\library\AbsListViewScrollDetector.cs:62
[MonoDroid] at Android.Widget.AbsListView/IOnScrollListenerInvoker.n_OnScroll_Landroid_widget_AbsListView_III (intptr,intptr,intptr,int,int,int) [0x00011] in /Users/builder/data/lanes/monodroid-mlion-monodroid-4.21-series/9e05e39f/source/monodroid/src/Mono.Android/platforms/android-21/src/generated/Android.Widget.AbsListView.cs:388
[MonoDroid] at (wrapper dynamic-method) object.dd6af828-f011-40ad-814e-b7d97a9b2f4a (intptr,intptr,intptr,int,int,int) <IL 0x00029, 0x0003f>
[MonoDroid]   --- End of managed exception stack trace ---
[MonoDroid] java.lang.NoSuchMethodError: animate
[MonoDroid]     at com.refractored.fab.AbsListViewScrollDetector.n_onScroll(Native Method)
[MonoDroid]     at com.refractored.fab.AbsListViewScrollDetector.onScroll(AbsListViewScrollDetector.java:30)
[MonoDroid]     at android.widget.AbsListView.invokeOnItemScrollListener(AbsListView.java:772)
[MonoDroid]     at android.widget.AbsListView.trackMotionScroll(AbsListView.java:3411)
[MonoDroid]     at android.widget.AbsListView.onTouchEvent(AbsListView.java:2233)
[MonoDroid]     at android.widget.ListView.onTouchEvent(ListView.java:3446)
[MonoDroid]     at android.view.View.dispatchTouchEvent(View.java:3885)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:903)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:942)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:942)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:942)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:942)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:942)
[MonoDroid]     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:942)
[MonoDroid]     at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:1691)
[MonoDroid]     at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1125)
[MonoDroid]     at android.app.Activity.dispatchTouchEvent(Activity.java:2096)
[MonoDroid]     at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:1675)
[MonoDroid]     at android.view.ViewRoot.deliverPointerEvent(ViewRoot.java:2194)
[MonoDroid]     at android.view.ViewRoot.handleMessage(ViewRoot.java:1878)
[MonoDroid]     at android.os.Handler.dispatchMessage(Handler.java:99)
[MonoDroid]     at android.os.Looper.loop(Looper.java:130)
[MonoDroid]     at android.app.ActivityThread.main(ActivityThread.java:3683)
[MonoDroid]     at java.lang.reflect.Method.invokeNative(Native Method)
[MonoDroid]     at java.lang.reflect.Method.invoke(Method.java:507)
[MonoDroid]     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:839)
[MonoDroid]     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:597)
[MonoDroid]     at dalvik.system.NativeStart.main(Native Method)
```

**Comment by jamesmontemagno** [comment]
Nice, I will take a look at this.

is there a reason you are still targeting 2.3?

**Comment by reknih** [comment]
I'm writing a schedule app for my school where some students still use some older / lower-tier Android devices. I target API Level 8 as a minimum - I haven't yet tested it on a 2.2 emulator, however.

**Comment by jamesmontemagno** [comment]
Now fixed in 1.3.0-beta1 nuget package.

I did not want to bundle in and require nineoldandroids so I am using the old animator and just changing visibility, which seems to work alright for that small percent that aren't on 4.0+
